hack/verify-shellcheck.sh

```diff
@@ -51,21 +48,6 @@ join_by() {
 SHELLCHECK_DISABLED="$(join_by , "${disabled[@]}")"
 readonly SHELLCHECK_DISABLED
 
-# creates the shellcheck container for later use
```

Comment:
I recommend just reading the new file at https://github.com/kubernetes/kubernetes/blob/60d50f4ce8d7559757100b274cc61bfdbbcfdcff/hack/verify-shellcheck.sh

basically this change is:
- we no longer need to lint file by file because we don't need to track which files failed
  - instead of creating a container to exec in repeatedly, just run a lint of all files at once one-shot in a container
     - do the equivalent when not running in a container
- we no longer need to check that the failures filed is alphabetized
- stop suggesting that people can add failures to the failures file, instead they can add a shellcheck disable comment if truly necessary